### PR TITLE
Adds a makeCreatedResponse to Request

### PR DESCRIPTION
**Pull request by grosch**
Adds a helper for `POST` type routes where the creation should return a proper 201 HTTP status code as well as including the `Location` HTTP header.  Notice the return type is `Response`, not the model object.

```swift
func create(req: Request) throws -> EventLoopFuture<Response> {
    let todo = try req.content.decode(Todo.self)
    return todo.create(on: req.db).flatMapThrowing {
       try req.makeCreatedResponse(for: todo)
    }
}
```

### CODE REVIEW

**Comment by 0xTim** [comment]
This will definitely need tests if Tanner wants to add this and it should probably use `Content` not `Model`

File: Sources/Vapor/Request/Request.swift [hunk 1]
```diff
@@ -186,4 +186,43 @@ public final class Request: CustomStringConvertible {
...
+    ///    }
+    /// }
+    /// ```
+    /// - Parameters:
+    ///   - obj: The newly created model.
+    ///   - mediaType: How to encode the response, defaulting to `.json`
+    ///   - customLocation: The location header to use. Defaults to the current route with the `obj` ID appended.
+    /// - Throws: If the object hasn't been created or the encoding fails.
+    /// - Returns: A `Response` object.
+    public func makeCreatedResponse<T>(for obj: T, as mediaType: HTTPMediaType = .json, at customLocation: String? = nil) throws -> Response where T: Model {
```
Comment (tanner0101) [inline Sources/Vapor/Request/Request.swift]: This would need to go in `vapor/fluent` to use the `Model` protocol.

Comment (tanner0101) [reply Sources/Vapor/Request/Request.swift]: A static extension on `Response` might be more concise:
```swift
Response.created(user, for: req)
// If you have type inference
.created(user, for: req)
```
As opposed to
```swift
req.makeCreatedResponse(for: user)

**Comment by grosch** [comment]
Will move to fluent and change to a static method on response as per @tanner0101 suggestion.

@0xTim It's purposefully on `Model` as opposed to `Content` so it can generate the `Location` header via the ID, which doesn't necessarily exist on `Content`

**Pull request closed without merging**